src/Autocomplete.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from 'react';
+import { Person } from './types/Person';
+
+interface AutocompleteProps {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+}
+
+export const Autocomplete: React.FC<AutocompleteProps> = ({
+  people,
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [suggestions, setSuggestions] = useState<Person[]>([]);
+  const [noSuggestions, setNoSuggestions] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      const trimmedInput = inputValue.trim();
+
+      if (trimmedInput === '') {
```

Comment:
According to the checklist, the filter function should not be called if the user entered only spaces. But in the current implementation, if the user enters spaces, the filter function will be called. To avoid this, you should check not only that the trimmedInput is not an empty string, but also that it is not composed entirely of spaces.